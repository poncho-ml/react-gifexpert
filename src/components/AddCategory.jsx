import { useState } from "react"

export const AddCategory = ( {onNewCategory} ) => {

    const [inputValue, setInputValue] = useState();
    const handleChange = ({target}) => {
        setInputValue(target.value);
    }
    const handleSubmit = (event) => {
        event.preventDefault();
        const newTrimmedValue = inputValue.trim();
        if(!inputValue || newTrimmedValue.length <= 1) return;
        // onNewCategory((categories) => [ inputValue, ...categories]);
        onNewCategory(newTrimmedValue);
        setInputValue('');  
    }

  return (
    <form onSubmit={handleSubmit} action=''>
        <input 
            type="text"
            placeholder="Buscar GIFs"
            value={inputValue || ''}
            onChange={handleChange}
        />
    </form>
  )
}
