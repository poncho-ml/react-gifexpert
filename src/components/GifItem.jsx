import PropTypes from 'react'

export const GifItem = ({title, url}) => {

    return (
    <div className="card">
        <img src={url} alt="" />
        <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  url: PropTypes.string,
  title: PropTypes.string
}
GifItem.defaultProps = {
    title: 'No hay t√≠tulo',
}