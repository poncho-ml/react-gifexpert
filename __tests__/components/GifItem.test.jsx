import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Testing GifItem', () => { 
    const title = 'Este es el tÃ­tulo';
    const url = 'https://th.bing.com/th/id/OIP.lvR5nni7xdPCxU-tC6tzzgHaHa?pid=ImgDet&rs=1';
    const dataObj = {
        title: title,
        url: url
    };

    test('should match the snapshot', () => { 
        const {container} = render( <GifItem {...dataObj} key='key' />);
        expect(container).toMatchSnapshot();
     })
})