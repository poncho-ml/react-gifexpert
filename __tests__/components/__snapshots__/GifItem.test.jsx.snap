// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt=""
      src="https://th.bing.com/th/id/OIP.lvR5nni7xdPCxU-tC6tzzgHaHa?pid=ImgDet&rs=1"
    />
    <p>
      Este es el t√≠tulo
    </p>
  </div>
</div>
`;
